frontend/src/metabase/dashboard/components/PublicLinkPopover/DashboardPublicLinkPopover.tsx

```diff
@@ -0,0 +1,49 @@
+import type { Dashboard } from "metabase-types/api";
+import { createPublicLink, deletePublicLink } from "metabase/dashboard/actions";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import { publicDashboard as getPublicDashboardUrl } from "metabase/lib/urls";
+import { getSetting } from "metabase/selectors/settings";
+import { PublicLinkPopover } from "./PublicLinkPopover";
+
+export const DashboardPublicLinkPopover = ({
+  dashboard,
+  target,
+  isOpen,
+  onClose,
+}: {
+  dashboard: Dashboard;
+  target: JSX.Element;
+  isOpen: boolean;
+  onClose: () => void;
+}) => {
+  const dispatch = useDispatch();
+
+  const uuid = dashboard.public_uuid;
+
+  const siteUrl = useSelector(state => getSetting(state, "site-url"));
```

Comment:
Also, I've commented about passing `siteUrl` to `metabase/lib/urls/*` before. I'm not sure why would we do that because when we don't pass it, the URL function would use the `site-url` settings via `MetabaseSettings.get("site-url")` anyway, so it feels redundant.

Reply:
I ended up doing that because MetabaseSettings just wasn't working with my tests. I figured that something was wrong and ended up passing the site-url directly

Reply:
You can use `mockSettings` which also sets the value in `MetabaseSettings` automatically, or just set `MetabaseSettings` value separately since they're not in sync by default in unit tests.
 https://github.com/metabase/metabase/blob/2a77c3041174167098a9f0e3eeebc2702e634789/frontend/test/__support__/settings.ts#L7

Reply:
Oh ok this is exactly what I was looking for. Thanks for the shout, I'll revert those changes as well. I can't say I liked them either